src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -230,24 +206,9 @@ on_exit() {
     set -xe
   fi
 
-  # backup cache
   if [[ $preserve_cache == true ]]; then
     echo "=== Execution completed — caching current session to $DX_CACHEDIR/$NXF_UUID"
-
-    # wrap cache folder and upload cache.tar
-    if [[ -n "$(ls -A .nextflow)" ]]; then
-      tar -cf cache.tar .nextflow
-
-      CACHE_ID=$(dx upload "cache.tar" --path "$DX_CACHEDIR/$NXF_UUID/cache.tar" --no-progress --brief --wait -p -r) &&
-        echo "Upload cache of current session as file: $CACHE_ID" &&
-        rm -f cache.tar ||
-        echo "Failed to upload cache of current session $NXF_UUID"
-    else
-      echo "No cache is generated from this execution. Skip uploading cache."
-    fi
-
-  # preserve_cache is false
-  # clean up files of this session
+    upload_session_cache_file
```

Comment:
what if both s3 and preserve_cache are true? 

Will this and other cmds (`if [[ $preserve_cache == true ]]`) still be triggered?

Reply:
Yes, it will (and should) be executed unconditionally if `if [[ $preserve_cache == true ]]`